### ISSUE TITLE
oidccl: TestOIDCEnabled is flaky on release-23.1

### ISSUE THREAD

**Issue by rafiss** [cockroachdb/cockroach#112146]
Seen here on this PR on release-23.1: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_BazelEssentialCi/12136051?expandBuildProblemsSection=true&hideProblemsFromDependencies=false&expandBuildTestsSection=true&hideTestsFromDependencies=false&expandBuildChangesSection=true

```
Failed
=== RUN   TestOIDCEnabled/login_redirect
    authentication_oidc_test.go:238:
          Error Trace:  github.com/cockroachdb/cockroach/pkg/ccl/oidcccl/authentication_oidc_test.go:238
          Error:        Not equal:
                        expected: "some/random/path"
                        actual  : "/"
                        Diff:
                        --- Expected
                        +++ Actual
                        @@ -1 +1 @@
                        -some/random/path
                        +/
          Test:         TestOIDCEnabled/login_redirect
```

Jira issue: CRDB-32254

**Comment by herkolategan** [comment]
Could be related to a change I made: https://github.com/cockroachdb/cockroach/pull/112129
Going to investigate.

**Comment by dhartunian** [comment]
@herkolategan I'll have a PR up shortly. the test doesn't work in tenant mode.

**Comment by herkolategan** [comment]
@dhartunian Ah thanks, I see my backport didn't go that far back, confused myself with the latest release branch just being cut.

**Comment by dhartunian** [comment]
Resolved via #112423
